packages/extension/src/background/handlers/Tabs.ts

```diff
@@ -72,6 +73,28 @@ export default class Tabs {
     return this.#state.sign(url, new RequestExtrinsicSign(request), { address, ...pair.meta });
   }
 
+  private rpcSend (request: RequestRpcSend): Promise<JsonRpcResponse> {
+    return this.#state.rpcSend(request);
+  }
+
+  private rpcSetProvider (request: RequestRpcProvider): Promise<null> {
+    return this.#state.rpcSetProvider(request);
+  }
+
+  private async rpcSubscribe (request: RequestRpcSubscribe, id: string, port: chrome.runtime.Port): Promise<boolean> {
+    const cb = createSubscription<'pub(accounts.subscribe)'>(id, port);
```

Comment:
just a thought, `id` is provided by the webpage (which can call `postMessage` manually, bypassing `extension-inject`), so a malicious page could provide the `id` of an existing subscription made by another page (iirc it's incremental so wouldn't be difficult) and it would overwrite the mapping in `subscriptions`. this would mess up our logic of unsubscribing to a page's subscriptions when the page closes.